sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/implementation/AzureEnvironment.java

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.implementation;
+
+import com.azure.core.experimental.http.policy.ARMChallengeAuthenticationPolicy;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An instance of this class describes an environment in Azure.
+ * This is a temporary impl class to support {@link ARMChallengeAuthenticationPolicy} until it
+ * moves to azure-resource-manager package.
+ */
+public final class AzureEnvironment {
+    /** the map of all endpoints. */
+    private final Map<String, String> endpoints;
```

Comment:
Should this data structure have more fields other than endpoints? Such as an identifier.

Reply:
not needed currently.